test/udp_sender.js

```diff
@@ -249,4 +250,20 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', () => {
+        sender._client.on('error', err => {
+            assert.equal(err.message, 'getaddrinfo ENOTFOUND foo.bar.com')
```

Comment:
> Also, I am not sure you would actually get uncaught error here since the default reporter buffers the spans.

I can confirm that my fix indeed solves the problem. Here is a simple reproducible example:

```js
var http = require('http')
var tracer = require('jaeger-client').initTracer({
    'serviceName': 'myService',
    'reporter': {
      logSpans: false,
      agentHost: 'foo.bar.com',
      flushIntervalMs: 10
    },
    'sampler': {
      host: 'foo.bar.com',
      param: 1,
      type: 'const'
    }
  })

http.createServer((req, res) => {
  span = tracer.startSpan('mySpan')
  res.end('response string')
  span.finish()
}).listen('8080')
```

```json
{
  "name": "example",
  "version": "1.0.0",
  "dependencies": {
    "jaeger-client": "3.5.3"
  }
}
```
```sh
$ npm install
$ node index.js
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: getaddrinfo ENOTFOUND foo.bar.com
    at errnoException (dns.js:28:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
```

Now to see the fix in action, navigate to:
```
node_modules/jaeger-client/dist/src/reporters/udp_sender.js
```
and add my fix to line 141:
```js
this._client.on('error', err => {
    console.log(`error sending span: ${err}`)
})
```
now restart the node server and hit `localhost:8080` and you will see the error is handled gracefully.